### PR TITLE
Update properties section of model details and version details

### PR DESCRIPTION
**Pull request by ppadti**
Closes: [RHOAIENG-6996](https://issues.redhat.com/browse/RHOAIENG-6996) and [RHOAIENG-13057](https://issues.redhat.com/browse/RHOAIENG-13057)

## Description
This PR aims to update Properties section of model details page and version details page:
  - Table expands when a user clicks on Add property button
  - New row appears in the properties table with the text input for Key field automatically gets the cursor focus
  
 The description input field of model details and version details pages also get cursor focus.
 Labels of model details and version details page are displayed without truncation.

https://github.com/user-attachments/assets/b9a18b8d-af43-4e47-9d1e-cccc177dd421

## How Has This Been Tested?
1. Try to add new row to Properties table with more than 5 property, so it will expand and the row will autofocus.
2. Make sure the description input field gets auto focus when we click on edit button of description section in model details and version details page.
3. Make sure the labels are not truncated any more in model details and version details page.

## Test Impact
Added and updated cypress tests.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by ppadti** [comment]
@yih-wang Could you please take a look at the above screenshots?

**Comment by yih-wang** [comment]
LGTM, thanks @ppadti !

**Review by manaswinidas — CHANGES_REQUESTED** [review]
I didn't find these in this PR while testing:
- The list should follow the creation order (oldest on the top, newest at the bottom), not the alphabetical order
- The fix for [RHOAIENG-13057](https://issues.redhat.com/browse/RHOAIENG-13057) - can't see the tooltip with full content for truncated labels
Otherwise lgtm.

**Comment by ppadti** [comment]
> I didn't find these in this PR while testing:
> 
> * The list should follow the creation order (oldest on the top, newest at the bottom), not the alphabetical order
> * The fix for [RHOAIENG-13057](https://issues.redhat.com/browse/RHOAIENG-13057) - can't see the tooltip with full content for truncated labels
>   Otherwise lgtm.

1. we can't sort it by creation order, because we only have key/value and decided to not to do it. [Mentioned here in the comment section.](https://issues.redhat.com/browse/RHOAIENG-6996?focusedId=25754903&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25754903)
2. we decided yesterday during standup to show the label without truncation, so no tooltip required. [comment mentioned](https://issues.redhat.com/browse/RHOAIENG-13057?focusedId=25799531&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25799531) here in the issue.

**Review by manaswinidas — APPROVED** [review]
Thanks @ppadti 

/lgtm

**Comment by manaswinidas** [comment]
@ConorOM1 test-id changes in this PR

File: frontend/src/components/EditableTextDescriptionListGroup.tsx [hunk 1]
```diff
@@ -36,6 +36,7 @@ const EditableTextDescriptionListGroup: React.FC<EditableTextDescriptionListGrou
       isSavingEdits={isSavingEdits}
       contentWhenEditing={
         <TextArea
+          autoFocus
```
Comment (mturley) [inline frontend/src/components/EditableTextDescriptionListGroup.tsx:39]: Nice, I wasn't aware of this prop

**Review by mturley — APPROVED** [review]
Good stuff, thanks @ppadti

**Pull request merged**